### PR TITLE
feat: allow deps to be added as workflows

### PR DESCRIPTION
**Pull request by kirederik**
this commit introduces a new flag to the `update dependencies` command:
`--as-workflow`.

when set, the dependencies will be added to the promise as a workflow, instead
of copying them all to the promise `spec.dependencies` field.

updating dependencies with `--as-workflow` requires `--image` to be set as well.

if there's a `dependencies.yaml` file in the working directory, it will be
removed; if there's a `promise.yaml` file, the contents of `spec.dependencies`
will be wiped out -- we can change this behavior if needed.

### CODE REVIEW

File: test/update_test.go [hunk 1]
```diff
@@ -311,6 +378,76 @@ var _ = Describe("update", func() {
 				})
 			})
 		})
+
+		Context("--as-workflow flag", func() {
```
Comment (richcooper95) [inline test/update_test.go:382]: Is this supposed to be inside the `Context("dependencies.yaml does not exist", ...)` on line 336?

File: cmd/update_dependencies.go [hunk 2]
```diff
@@ -23,44 +23,43 @@ Kratix update dependencies local-dir/ `,
 	RunE: updateDependencies,
 }
 
+var depsAsWorkflow bool
+
 func init() {
 	updateCmd.AddCommand(updateDependenciesCmd)
 	updateDependenciesCmd.Flags().StringVarP(&dir, "dir", "d", ".", "Directory to read Promise from")
+	updateDependenciesCmd.Flags().BoolVar(&depsAsWorkflow, "as-workflow", false, "Whether to include dependencies as a workflow or not; default is false")
+	updateDependenciesCmd.Flags().StringVarP(&image, "image", "i", "", "The image used by this container.")
```
[the remark this answers is not in the document]
Comment (kirederik) [reply cmd/update_dependencies.go:32]: maybe it should fail if its provided but no --as-workflow? or perhaps we could use the presence of `--image` to determine if it should be added as a workflow?

[the remark this answers is not in the document]
Comment (kirederik) [reply cmd/update_dependencies.go:32]: --image is what we use in the `add container` command; i would like them to be the same

File: cmd/update_dependencies.go [hunk 3]
```diff
@@ -140,3 +139,80 @@ func getPromise(filePath string) (v1alpha1.Promise, error) {
...
+		err = os.Remove(filepath.Join(dir, dependenciesFileName))
+	case promiseFileName:
+		err = updatePromiseDependencies([]v1alpha1.Dependency{})
+	}
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("Dependencies added as a Promise workflow.")
+	fmt.Println("Don't forget to build and push the image.")
```
[the remark this answers is not in the document]
Comment (kirederik) [reply cmd/update_dependencies.go:190]: i hope we end up with some kind of `kratix build container` command so let's not try to build it here;

not opposed to adding a command to the output though

File: cmd/update_dependencies.go [hunk 4]
```diff
@@ -140,3 +139,80 @@ func getPromise(filePath string) (v1alpha1.Promise, error) {
...
+				return err
+			}
+		}
+		fileContents, err := os.ReadFile(filepath.Join(src, f.Name()))
+		if err != nil {
+			return err
+		}
+		if err := os.WriteFile(filepath.Join(dest, f.Name()), fileContents, 0644); err != nil {
+			return err
+		}
```
[the remark this answers is not in the document]
Comment (kirederik) [reply cmd/update_dependencies.go:215]: i think what i missed was a simple "continue" at the end of the `if f.isDir()`?

File: test/update_test.go (hunk 1 above)
Comment (kirederik) [reply test/update_test.go:382]: i don't think so

File: cmd/update_dependencies.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (richcooper95) [reply cmd/update_dependencies.go:32]: Yeah, I think any of those would work! Mild preference for having `--image` as the only flag, and using that to toggle whether we write to the workflow or not

[the remark this answers is not in the document]
Comment (richcooper95) [reply cmd/update_dependencies.go:32]: In that case, WDYT about changing to `image-name` in both? It's clearer that we're building an image of that name; `--image` feels like you're providing a pre-built image

File: cmd/update_dependencies.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (richcooper95) [reply cmd/update_dependencies.go:215]: That works too! (And removes the scary 👻 `else` 👻)

File: cmd/update_dependencies.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (kirederik) [reply cmd/update_dependencies.go:32]: not sure; i think `--image` is fine (it's also what kubectl uses for their commands so not that foreign). i suggest we leave it as is and maybe have a separate conversation/thread about the flag name?

File: cmd/update_dependencies.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (kirederik) [reply cmd/update_dependencies.go:215]: #noElseTeam

**Review by richcooper95 — APPROVED** [review]

File: cmd/update_dependencies.go [hunk 5]
```diff
@@ -205,11 +201,16 @@ func copyFiles(src, dest string) error {
 			if err := copyFiles(filepath.Join(src, f.Name()), filepath.Join(dest, f.Name())); err != nil {
 				return err
 			}
+
+			continue
 		}
+
+		fmt.Println("Reading file: " + f.Name())
```
Comment (kirederik) [inline cmd/update_dependencies.go:208]: debug line?

File: cmd/update_dependencies.go [hunk 6]
```diff
@@ -205,11 +201,16 @@ func copyFiles(src, dest string) error {
...
 		fileContents, err := os.ReadFile(filepath.Join(src, f.Name()))
 		if err != nil {
 			return err
 		}
+		fmt.Println("Read file: " + f.Name())
```
Comment (kirederik) [inline cmd/update_dependencies.go:213]: debug line?

**Pull request merged**